src/emit/expression.rs

```diff
@@ -1349,10 +1349,15 @@ pub(super) fn expression<'a, T: TargetRuntime<'a> + ?Sized>(
 
             let start = unsafe { bin.builder.build_gep(data, &[offset], "start") };
 
-            if let Type::Bytes(n) = &returns[0] {
+            if matches!(returns[0], Type::Bytes(_) | Type::FunctionSelector) {
+                let n = match &returns[0] {
+                    Type::Bytes(n) => *n,
+                    Type::FunctionSelector => ns.target.selector_length(),
+                    _ => unreachable!(),
+                };
```

Comment:
How about `returns[0].bits()`?

Reply:
Still better than open-coding it. You could also also add an `fn bytes(self, ns: &Namespace)` to `Type`.

Reply:
Fixed